### PR TITLE
chore(agw): implement shared mconfig and use it for Sentry config

### PR DESCRIPTION
**Pull request by alexzurbonsen**
## Summary

As part of #9508 all the Sentry config for Python services should be migrated to shared mconfig.
- implement loading of shared mconfig (follow up to #9964)
- use it to initialize Sentry SDK
- remove Sentry config from control_proxy.yml

## Test Plan

Locally tested that config can be read.
Included shared_mconfig in respective unit test.

### CODE REVIEW

File: lte/gateway/configs/control_proxy.yml
Comment (themarwhal) [inline lte/gateway/configs/control_proxy.yml]: Let's keep these in place because
1. We have people who use it still
2. We still want to be able to override the values on a per-agw basis. (This is probably useful while we're testing out all the Sentry enablement changes. Once we can confirm we've gotten rid of noise and any performance risks, we can just have the configuration be all or nothing per-network.)

So I would modify the PR to always take the control_proxy value as a prescendence if it's not empty during this migration phase.

Comment (alexzurbonsen) [reply lte/gateway/configs/control_proxy.yml]: Good point. Will take care of it.

Comment (alexzurbonsen) [reply lte/gateway/configs/control_proxy.yml]: @themarwhal The PR now reads the config from control_proxy.yml first (if sentry_url is present) :)

File: orc8r/gateway/python/magma/common/sentry.py
Comment (themarwhal) [inline orc8r/gateway/python/magma/common/sentry.py]: do we need this else if we're returning in the if condition above?

File: orc8r/gateway/python/magma/common/sentry.py
Comment (themarwhal) [inline orc8r/gateway/python/magma/common/sentry.py]: Can we add some more detail on where this value can be placed? (So we can say what's described in the function doc above somewhere. control_proxy config for per-agw override and mconfig.shared_config.sentry_config for network wide configs)

Comment (themarwhal) [reply orc8r/gateway/python/magma/common/sentry.py]: Hmm actually, maybe it is too noisy if we print that for every service

File: orc8r/gateway/python/magma/configuration/mconfig_managers.py
Comment (themarwhal) [inline orc8r/gateway/python/magma/configuration/mconfig_managers.py:195]: Can we suppress this pep8 warning and say the docs are in the abstract method

File: orc8r/gateway/python/magma/common/sentry.py
Comment (alexzurbonsen) [reply orc8r/gateway/python/magma/common/sentry.py]: True, it might be... Do you think printing the single line, as we currently do, might be too noisy as well? (On the other hand it may be helpful for users, who wonder why sentry isn't working, even though they enabled it in the service config.)

File: orc8r/gateway/python/magma/configuration/mconfig_managers.py
Comment (alexzurbonsen) [reply orc8r/gateway/python/magma/configuration/mconfig_managers.py:195]: Disabled the missing-docstrings warning for the entire class since all methods are already documented in base class. Would that be a reasonable alternative?

File: orc8r/gateway/python/magma/common/sentry.py
Comment (themarwhal) [reply orc8r/gateway/python/magma/common/sentry.py]: Yeah since we already have the single line message and that doesn't seem to bother us, let's just make it a bit more descriptive then :) You make a good point about there being an additional config about enabling / disabling sentry so it is confusing.

**Review by themarwhal — APPROVED** [review]

File: orc8r/gateway/python/magma/common/sentry.py
Comment (themarwhal) [inline orc8r/gateway/python/magma/common/sentry.py]: 👍

Comment (alexzurbonsen) [reply orc8r/gateway/python/magma/common/sentry.py]: Pushed one more time to make format check happy (and also markdown lint check seems to be broken on master. ran make precommit_fix in docs and included changes here - only `contribute_id_mappings.md` is affected).

**Review by ardzoht — COMMENTED** [review]
`lte/gateway` changes lgtm

**Comment by hcgatewood** [comment]
cc @christinewang5

**Review by hcgatewood — COMMENTED** [review]
Looking good! I just skimmed, but didn't see anywhere that we were handling restarting a service based on its shared mconfig changing

I'm assuming we want to handle that somewhere -- or maybe I missed it? cc @themarwhal

**Comment by themarwhal** [comment]
> Looking good! I just skimmed, but didn't see anywhere that we were handling restarting a service based on its shared mconfig changing
> 
> I'm assuming we want to handle that somewhere -- or maybe I missed it? cc @themarwhal

Yes no service restart on shared config update for now. It's not that crucial for Sentry config at this current stage of testing, but we will get it in eventually :) I think I filed an issue for this, will dig that up or create a new one and link it here for future context.

EDIT: Also wanted to separate out the restart logic from just reading the config as the restart logic might require a bit more thought.

cc @hcgatewood 

cc @alexzurbonsen Do you mind adding a TODO item inside the mconfig handling logic to make it clear that it's pending work?

**Comment by alexzurbonsen** [comment]
> cc @alexzurbonsen Do you mind adding a TODO item inside the mconfig handling logic to make it clear that it's pending work?

@themarwhal Of course, added a TODO in process_update of config_manager.py

**Comment by themarwhal** [comment]
I think this PR only needs a review from either @pshelar or @koolzz

**Review by koolzz — APPROVED** [review]

**Review by themarwhal — APPROVED** [review]

**Review by ardzoht — APPROVED** [review]

**Comment by themarwhal** [comment]
Hey @alexzurbonsen ! This PR is good to go now, let me know if you want me to merge it. (Or label `ready2merge`)

**Comment by alexzurbonsen** [comment]
> Hey @alexzurbonsen ! This PR is good to go now, let me know if you want me to merge it. (Or label `ready2merge`)

Great. Yes, @themarwhal feel free to merge it. Thanks everyone :)

**Pull request merged**
